nodejs/awsx/apigateway/api.ts

```diff
@@ -625,13 +729,20 @@ function addStaticRouteToSwaggerSpec(
         // Take whatever path the client wants to host this folder at, and add the
         // greedy matching predicate to the end.
         const proxyPath = directoryServerPath + "{proxy+}";
-        addSwaggerOperation(swagger, proxyPath, swaggerMethod("ANY"), createSwaggerOperationForObjectKey(directoryKey, role, "proxy"));
+        const swaggerOperation = createSwaggerOperationForObjectKey(directoryKey, role, "proxy");
+        if (directory.requiredParameters) {
+            addRequiredParametersToSwaggerOperation(swaggerOperation, directory.requiredParameters);
+        }
+        if (directory.requestValidator) {
+            swaggerOperation["x-amazon-apigateway-request-validator"] = directory.requestValidator;
+        }
+        addSwaggerOperation(swagger, proxyPath, swaggerMethod("ANY"), swaggerOperation);
     }
 
     function createSwaggerOperationForObjectKey(
-            objectKey: string,
-            role: aws.iam.Role,
-            pathParameter?: string): SwaggerOperation {
+        objectKey: string,
+        role: aws.iam.Role,
+        pathParameter?: string): SwaggerOperation {
```

Comment:
note: the indentation above was intentional.  the problem with single-indent on parameters is it can make it more difficult to distinguish the parameters from the body statements.  

Not pushing back hard on htis.  if you prefer this form, that's ok.  if so though, please leave the blank line between the parameters and first statements.

Reply:
my editor automatically formatted this. i can def change it back